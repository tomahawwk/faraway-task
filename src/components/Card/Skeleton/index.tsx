import Skeleton from 'react-loading-skeleton';

const SkeletonCard = ({...props}) => {
  return (
    <div className="skeleton-card h-[250px] p-sm">
      <Skeleton
        count={6}
        {...props}
        baseColor="#0a0a0a"
        highlightColor="#141414"
      />
    </div>
  );
};

export default SkeletonCard;
