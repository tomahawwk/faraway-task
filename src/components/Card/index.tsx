import {FC} from 'react';
import {ICharacter} from '../../store/models/ICharacter';
import {Link} from 'react-router-dom';
import LazyImage from 'ui-kit/LazyImage';

const Card: FC<ICharacter> = ({name, url}) => {
  return (
    <Link
      to={url.match(/\/(\d+)\/$/)?.[1] || ''}
      className="p-sm grid w-full h-[250px] gap-sm rounded-sm bg-secondary-dark duration-300
      group hover:scale-[1.04] hover:bg-secondary-main">
      <div className="flex flex-col gap-xs will-change-transform justify-between">
        <div className="flex items-center gap-xs">
          <div className="w-[40px] h-[40px] rounded-full bg-grey relative">
            <LazyImage
              src="images/user.png"
              className="absolute top-0 bottom-0 left-0 right-0 m-auto w-[24px]"
            />
          </div>
          <div className="text-[20px] font-medium leading-6 line-clamp-1">
            {name}
          </div>
        </div>
        <div className="flex pb-xs relative">
          <div>
            <img
              src="images/saber.png"
              alt="Explore more"
              className="w-[50px] opacity-50 duration-300 group-hover:opacity-100 will-change-transform"
            />
            <span
              className="absolute left-[50px] top-[2px] h-[4px] w-[100px] bg-primary-main rounded-[4px]
              origin-left scale-x-0 duration-300 group-hover:scale-x-100"
            />
          </div>
        </div>
      </div>
    </Link>
  );
};

export default Card;
