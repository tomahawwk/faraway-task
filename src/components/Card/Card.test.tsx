import {render, screen} from '@testing-library/react';
import Card from '.';
import {MemoryRouter} from 'react-router-dom';

describe('Card component', () => {
  const character = {
    name: 'Test Character',
    url: `/api/character/1`,
    birth_year: '',
    eye_color: '',
    mass: '',
    gender: '',
    hair_color: '',
    height: '',
    id: '',
    created: '',
  };

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(() => ({
      push: jest.fn(),
    })),
  }));

  test('renders character name correctly', () => {
    render(
      <MemoryRouter>
        <Card {...character} />
      </MemoryRouter>,
    );

    expect(screen.getByText(/Test Character/)).toBeInTheDocument();
  });

  test('renders a saber image with the correct src', () => {
    render(
      <MemoryRouter>
        <Card {...character} />
      </MemoryRouter>,
    );

    const saberImageElement = screen.getByRole('img', {name: /Explore more/i});
    expect(saberImageElement).toHaveAttribute('src', 'images/saber.png');
  });

  test('Card snapshot', () => {
    const card = render(
      <MemoryRouter>
        <Card {...character} />
      </MemoryRouter>,
    );

    expect(card).toMatchSnapshot();
  });
});
