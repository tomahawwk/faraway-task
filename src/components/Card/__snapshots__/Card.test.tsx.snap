// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="p-sm grid w-full h-[250px] gap-sm rounded-sm bg-secondary-dark duration-300
 group hover:scale-[1.04] hover:bg-secondary-main"
        href="/"
      >
        <div
          class="flex flex-col gap-xs will-change-transform justify-between"
        >
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="w-[40px] h-[40px] rounded-full bg-grey relative"
            >
              <span
                class=" lazy-load-image-background blur"
                style="color: transparent; display: inline-block;"
              >
                <img
                  class="absolute top-0 bottom-0 left-0 right-0 m-auto w-[24px]"
                  src="images/user.png"
                />
              </span>
            </div>
            <div
              class="text-[20px] font-medium leading-6 line-clamp-1"
            >
              Test Character
            </div>
          </div>
          <div
            class="flex pb-xs relative"
          >
            <div>
              <img
                alt="Explore more"
                class="w-[50px] opacity-50 duration-300 group-hover:opacity-100 will-change-transform"
                src="images/saber.png"
              />
              <span
                class="absolute left-[50px] top-[2px] h-[4px] w-[100px] bg-primary-main rounded-[4px]
 origin-left scale-x-0 duration-300 group-hover:scale-x-100"
              />
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="p-sm grid w-full h-[250px] gap-sm rounded-sm bg-secondary-dark duration-300
 group hover:scale-[1.04] hover:bg-secondary-main"
      href="/"
    >
      <div
        class="flex flex-col gap-xs will-change-transform justify-between"
      >
        <div
          class="flex items-center gap-xs"
        >
          <div
            class="w-[40px] h-[40px] rounded-full bg-grey relative"
          >
            <span
              class=" lazy-load-image-background blur"
              style="color: transparent; display: inline-block;"
            >
              <img
                class="absolute top-0 bottom-0 left-0 right-0 m-auto w-[24px]"
                src="images/user.png"
              />
            </span>
          </div>
          <div
            class="text-[20px] font-medium leading-6 line-clamp-1"
          >
            Test Character
          </div>
        </div>
        <div
          class="flex pb-xs relative"
        >
          <div>
            <img
              alt="Explore more"
              class="w-[50px] opacity-50 duration-300 group-hover:opacity-100 will-change-transform"
              src="images/saber.png"
            />
            <span
              class="absolute left-[50px] top-[2px] h-[4px] w-[100px] bg-primary-main rounded-[4px]
 origin-left scale-x-0 duration-300 group-hover:scale-x-100"
            />
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
